scripts/deposit.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable indent */
+const SnappBase = artifacts.require("SnappBase")
+
+const zero_address = "0x0000000000000000000000000000000000000000"
+const one_hash = "0x0000000000000000000000000000000000000001"
+
+module.exports = async (callback) => {
+    const instance = await SnappBase.deployed()
+    const [accountId, tokenId, amount] = await process.argv.slice(4)
+
+    const depositor = await instance.accountToPublicKeyMap.call(accountId)
+    if (depositor == zero_address) {
+        console.log("No account registerd at index %s", accountId)
+        callback()
+    }
+    const slot = (await instance.depositSlot.call()).toNumber()
+    const tx = await instance.deposit(tokenId, amount, {from: depositor})
```

Comment:
You might also wanna check that the user has the funds available and throw overwise..

Where do you approve the token transfer? IF it is only approved in setup_environment, you might need to say this at the beginning of the file. I would recommend doing a seperate approval, so that the deposit script can be used independently

Reply:
I can separate them and make three scripts:

- setup
- approve
- setup-and-approve

Reply:
I don't fully understand what you mean by deposit script being used independently... If we don't setup environment, we will also need a register token, open account, and a bunch of stuff.... I guess we will need these anyways. I really need to figure out how to import files.

Reply:
Okay, it would be for later for sure. But I could come up with some use case. E.g. a bot has already an account and he just wants to deposit some tokens ....

Reply:
I have included pre-validation in the deposit request!